index.js

```diff
@@ -18,28 +18,27 @@ module.exports.init = function (options, callback) {
   if (!options.secret) options.secret = process.env.ARTSY_SECRET;
 
   // Fetch the xapp token, cache, and refresh
-  fetchAndCacheToken(options)(callback);
+  fetchAndCacheToken(options, callback);
 }
 
-var fetchAndCacheToken = function (options){
-  return function(callback) {
-
-    // Get the token
-    request
-      .get(options.url + '/api/v1/xapp_token')
-      .query({ client_id: options.id, client_secret: options.secret })
-      .end(function(err, res) {
-        if (err) {
-          module.exports.emit('error', err);
-          if (callback) callback(err);
-          return
-        }
-        module.exports.token = res.body.xapp_token;
-        if (callback) callback(null, module.exports.token);
-
-        // Recurse this function to refresh the token it before it expires
-        var expiresAt = new Date(res.body.expires_in).getTime();
-        setTimeout(fetchAndCacheToken(options), (expiresAt - 1000) - Date.now());
-      });
-    }
+var fetchAndCacheToken = function (options, callback) {
+  // Get the token
+  request
+    .get(options.url + '/api/v1/xapp_token')
+    .query({ client_id: options.id, client_secret: options.secret })
```

Comment:
Does this have to be `get` with a `query` vs a `post` with a body? Not sure how sensitive this is but the id and secret will persist in the logs.

Reply:
I think so, this is a GET endpoint in Gravity.

Reply:
Good call though, that might be a security concern.  I guess if we wanted to address that we'd need to offer this via POST as well (from Gravity), and then we could switch to using that.

Reply:
When we were auditing our code earlier this year after the incident I had to fix a few other `get` + `query`'s that were leaking, but in those instances what they were leaking was, um..

Reply:
Is it possible that this overlaps with this bug ticket?: https://artsyproduct.atlassian.net/browse/GROW-1280

Reply:
I _think_ that ticket is about something a little bit different (wanting to keep credentials from appearing in URL's on the front-end). But the intent (on keeping it out of logs/URLs) is probably the same!

Reply:
Right, I'm suggesting that we'd want to use `post` for sending any sensitive information by default based on similar efforts elsewhere.

Reply:
Oh yea, true. I guess we should have a card about generally doing a sweet in Gravity and making sure all the endpoints like these are using POST bodies where possible.